pkg/receive/handler.go

```diff
@@ -397,6 +417,20 @@ func (h *Handler) receiveHTTP(w http.ResponseWriter, r *http.Request) {
 
 	tLogger := log.With(h.logger, "tenant", tenant)
 
+	// Impose limits only if Receive is in Router or RouterIngestor mode.
+	if h.receiverMode == RouterOnly || h.receiverMode == RouterIngestor {
```

Comment:
Should we inject the series limiter as a dependency into the handler? This way in the handler we always call `isUnderLimit` and the `nopHandler` will always return false. 

Ideally I would like to have this `if` decision made in one place, and we already do it here: https://github.com/thanos-io/thanos/pull/5520/files#diff-3d45f9beb5b670e94fe2715e21477a3fb8f5db7bf97f01cddf6e108bb98761a6R301-R302

Reply:
Have implemented this here: https://github.com/thanos-io/thanos/pull/5520/commits/d101a6160acfb6059703d605ff4b472f25059a45